src/generic/stage1/buf_block_reader.h

```diff
@@ -6,6 +6,9 @@ struct buf_block_reader {
   really_inline size_t block_index();
   really_inline bool has_full_block() const;
   really_inline const uint8_t *full_block() const;
+  really_inline bool has_remainder() const {
```

Comment:
I don't mind ending up with a different coding guideline, but I've been separating declaration from definition in more and more classes, including this one, so this method now sticks out :)

Reply:
Ok. Removing.

Reply:
Thanks. Note, I'm seriously not attached to the decision to move all function definitions out of the class. I know there are at least some (maybe most?) projects that declare one-liners should always be in the class. Some need to move to avoid header interdependency though, and some are too large and distract from one's ability to understand the interface. I moved them all so that I didn't have to think hard about where to put things--I am of the Feynman "buy 7 copies of the same clothes so you don't have to waste mental energy picking what to wear in the morning" persuasion.

Reply:
Just removing the function simplified the code, so that's fine.